development/contracts/ForkingManager.sol

```diff
@@ -0,0 +1,265 @@
+        address _globalExitRoot,
+        uint256 _arbitrationFee
+    ) external initializer {
+        zkEVM = _zkEVM;
+        bridge = _bridge;
+        forkonomicToken = _forkonomicToken;
+        parentContract = _parentContract;
+        globalExitRoot = _globalExitRoot;
+        arbitrationFee = _arbitrationFee;
+        ForkableUUPS.initialize(address(this), _parentContract, msg.sender);
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+     * @notice function to initiate and execute the fork
+     * @param _disputeData the dispute contract and call to identify the dispute
+     * @param newImplementations the addresses of the new implementations that will
+     * be used to create the second children of the contracts
+     */
+    function initiateFork(
+        DisputeData memory _disputeData,
+        NewImplementations calldata newImplementations
+    ) external onlyBeforeForking {
+        // Charge the forking fee
+        IERC20(forkonomicToken).safeTransferFrom(
+            msg.sender,
+            address(this),
+            arbitrationFee
+        );
+        // Create the children of each contract
+        NewInstances memory newInstances;
+        (
+            newInstances.forkingManager.one,
+            newInstances.forkingManager.two
+        ) = _createChildren(newImplementations.forkingManagerImplementation);
+        (newInstances.bridge.one, newInstances.bridge.two) = IForkableBridge(
+            bridge
+        ).createChildren(newImplementations.bridgeImplementation);
+        (newInstances.zkEVM.one, newInstances.zkEVM.two) = IForkableZkEVM(zkEVM)
+            .createChildren(newImplementations.zkEVMImplementation);
+        (
+            newInstances.forkonomicToken.one,
+            newInstances.forkonomicToken.two
+        ) = IForkonomicToken(forkonomicToken).createChildren(
+            newImplementations.forkonomicTokenImplementation
+        );
+        (
+            newInstances.globalExitRoot.one,
+            newInstances.globalExitRoot.two
+        ) = IForkableGlobalExitRoot(globalExitRoot).createChildren(
+            newImplementations.globalExitRootImplementation
+        );
+
+        // Initialize the zkEVM contracts
+        IPolygonZkEVM.InitializePackedParameters
+            memory initializePackedParameters;
+
+        {
+            // retrieve some information from the zkEVM contract
+            bytes32 genesisRoot = IPolygonZkEVM(zkEVM).batchNumToStateRoot(
+                IPolygonZkEVM(zkEVM).lastVerifiedBatch()
+            );
+            // the following variables could be used to save gas, but it requires via-ir in the compiler settings
+            string memory trustedSequencerURL = IPolygonZkEVM(zkEVM)
+                .trustedSequencerURL();
+            string memory networkName = IPolygonZkEVM(zkEVM).networkName();
+            // string memory version = "0.1.0"; // Todo: get version from zkEVM, currently only emitted as event
```

Comment:
I am not sure whether this is worth any effort. I will leave it as a todo for now